4_Zadania_dodatkowe/8.Parallax/js/app.js

```diff
@@ -0,0 +1,46 @@
+$(document).ready(function () {
+
+    var oldMousePositionX = 0;
+    var oldMousePositionY = 0;
+
+    var currentMousePositionX = 0;
+    var currentMousePositionY = 0;
+
+    var mouseMoveX = 0;
+    var mouseMoveY = 0;
+
+    $('.element')
+        .each(function() {
+            $(this).css('z-index', $(this).data('z'))
+        })
+        .each(function() {
+            $(this).css('left', $(this).data('x'))
+        })
+        .each(function() {
+            $(this).css('top', $(this).data('y'))
+        });
+
+    $('.scene').on('mouseenter', function (event) {
+        oldMousePositionX = event.offsetX;
+        oldMousePositionY = event.offsetY;
+    });
+
+    $('.scene').on('mousemove', function (event) {
```

Comment:
regarding `Punkt 6`:
1. notice `event` parameter in line 28
2. `even.target` should be an element that a cursor points at at the moment
3. they want you to check if `even.target` is one of `$('.element')`
4. if it's true then you're supposed to change `currentMousePositionX` and `currentMousePositionY` (add `left` and `top` css values from `even.target`)
5. I think it should be done before line 32

I didn't play with it, didn't test the code…